### ISSUE TITLE
Newer version do not work behind networks with proxy

### ISSUE THREAD

**Issue by dnoliver** [zrrrzzt/html-validator#84]
First, nice work! :D 

Now, the issue: Unfortunately, our network administrators have blessed us with the benefits of a web proxy  (for http and https) to access sites that are external to our company. We always run into issues that are caused by the proxy settings in the developer machine...

The first time I set the html validator job, it was working fine. The version at that point was 3.0.1.

Some days ago, this job started failing :( and the problem was ... the proxy!
For some reason, the html-validator was unable to communicate with the validation service through the corporate proxy.  Forcing to use the version 3.0.1 make it work again.

But, maybe you could check what was changed between 3.0.1 and latest 3.0.x that made that proxy support go bad. Maybe it is the request library update? 

I could provide logs, if you need them :)

**Comment by zrrrzzt** [comment]
The latest changes are available from the [releases page](https://github.com/zrrrzzt/html-validator/releases)

In v3.0.4 I switched from request to axios. Maybe axios has different default settings when it comes to proxies? Does v3.0.3 work for you?

I can take a look at the logs and see if I spot anything obvious.

**Comment by dnoliver** [comment]
I think that is the problem :)

Request library reacts to `http_proxy` `https_proxy` and `no_proxy` env vars 
Axios seems to not react to them. There is an issue created in the Axios Github repo here https://github.com/axios/axios/issues/1371

I think that the reason behind switching between one library and the other is because Axios is smaller than Request, so I support your decision here, but that make this package unusable behind a corporate proxy.

I already saw this issue with the node yeoman generator, it decided to use a non-Request library, and it made it unusable behind a proxy. 

Some solution I saw in other modules is to provide the option to set up the http library to do the request, something like:

```
var htmlValidator = require('html-validator')

// Using axios
var axiosBasedValidator = htmlValidator.createValidator({
  httpImplementation: require('axios')
});

// Using Request
var requestBasedValidator = htmlValidator.createValidator({
  httpImplementation: require('request')
});

// so this will work under proxy conditions
requestBasedValidator.run(file)

// And this not, but it will depend on smaller things
axiosBasedValidator.run(file)
```

I could try the v3.0.3 version later, but I think the switch of the http library caused the problem
